src/main/java/com/mparticle/kits/ButtonKit.java

```diff
@@ -73,165 +91,222 @@ public ButtonKit getInstance() {
+        if (e < 0) {
+            logError(currCode + " is not a valid currency code. Please conform to ISO4217.");
+            return null;
+        }
+        double total = 0.00;
+        for (Product product : event.getProducts()) {
+            total += product.getUnitPrice();
         }
-        if (referrer.equals(mStorage.getReferrer())) return;
-        mStorage.setReferrer(referrer);
-        final Map<String, String> attributes = getIntegrationAttributes();
-        attributes.put(ATTRIBUTE_REFERRER, referrer);
-        ButtonLog.verboseFormat(TAG, "Updated integration attributes to: %s", attributes);
-        setIntegrationAttributes(attributes);
+        long orderTotal = Math.round(total * Math.pow(10, e));
+
+        return new Order.Builder(orderId)
+                .setCurrencyCode(currCode)
+                .setAmount(orderTotal)
+                .build();
     }
 
+    /*
+     * Overrides for IdentityListener
+     */
+
     @Override
-    public List<ReportingMessage> onActivityCreated(final Activity activity, final Bundle savedInstanceState) {
-        return null;
+    public void onIdentifyCompleted(MParticleUser mParticleUser,
+            FilteredIdentityApiRequest filteredIdentityApiRequest) {
+
     }
 
     @Override
-    public List<ReportingMessage> onActivityStarted(final Activity activity) {
-        Intent intent = activity.getIntent();
-        if (intent == null) {
-            return null;
-        }
-        return handleIntent(intent.getData(), intent.getAction());
+    public void onLoginCompleted(MParticleUser mParticleUser,
+            FilteredIdentityApiRequest filteredIdentityApiRequest) {
+
     }
 
     @Override
-    public List<ReportingMessage> onActivityResumed(final Activity activity) { return null; }
+    public void onLogoutCompleted(MParticleUser mParticleUser,
+            FilteredIdentityApiRequest filteredIdentityApiRequest) {
+        clearAllData();
+    }
 
     @Override
-    public List<ReportingMessage> onActivityPaused(final Activity activity) { return null; }
+    public void onModifyCompleted(MParticleUser mParticleUser,
+            FilteredIdentityApiRequest filteredIdentityApiRequest) {
+
+    }
 
     @Override
-    public List<ReportingMessage> onActivityStopped(final Activity activity) { return null; }
+    public void onUserIdentified(MParticleUser mParticleUser) {
+
+    }
+
+    /*
+     * Overrides for ActivityListener
+     */
 
     @Override
-    public List<ReportingMessage> onActivitySaveInstanceState(final Activity activity, final Bundle outState) { return null; }
+    public List<ReportingMessage> onActivityCreated(Activity activity, Bundle bundle) {
+        trackIncomingIntent(activity.getIntent());
+        return null;
+    }
 
     @Override
-    public List<ReportingMessage> onActivityDestroyed(final Activity activity) { return null; }
+    public List<ReportingMessage> onActivityStarted(Activity activity) {
+        return null;
```

Comment:
We should call `trackIncomingIntent()` in both `onActivityCreated()` and `onActivityStarted()` to account for the case where the merchant app is open but in the background and a new deep link is sent from the publisher app.